packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts

```diff
@@ -0,0 +1,172 @@
+  }
+  const checkNode = {
+    ...node,
+    $type: node.$type || nodeType
+  } as typeof node;
+
+  // TODO in the following code, `typeof checkNode.$value === 'string' ? checkNode.$value :` is defined to please Jest TS compilation. It should be removed when supported
+  switch (checkNode.$type) {
+    case 'color':
+    case 'number':
+    case 'duration':
+    case 'fontWeight':
+    case 'fontFamily':
+    case 'dimension': {
+      return checkNode.$value.toString();
+    }
+    case 'strokeStyle': {
+      return renderCssTypeStrokeStyleValue(checkNode.$value);
+    }
+    case 'cubicBezier': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        checkNode.$value.join(', ');
+    }
+    case 'border': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.width} ${renderCssTypeStrokeStyleValue(checkNode.$value.style)} ${checkNode.$value.color}`;
+    }
+    case 'gradient': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `linear-gradient(0deg, ${checkNode.$value.map(({color, position}) => `${color} ${position}`).join(', ')})`;
+    }
+    case 'shadow': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.offsetX} ${checkNode.$value.offsetY} ${checkNode.$value.blur}  ${checkNode.$value.spread} ${checkNode.$value.color}`;
+    }
+    case 'transition': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        typeof checkNode.$value.timingFunction === 'string' ? checkNode.$value.timingFunction : checkNode.$value.timingFunction.join(' ') +
+          ` ${checkNode.$value.duration} ${checkNode.$value.delay}`;
+    }
+    case 'typography': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.fontWeight} ${checkNode.$value.fontFamily} ${checkNode.$value.fontSize} ${checkNode.$value.letterSpacing} ${checkNode.$value.lineHeight}`;
+    }
+    default: {
+      throw new Error(`Not supported type ${(checkNode as any).$type || 'unknown'} (value: ${(checkNode as any).$value || 'unknown'})`);
+    }
+  }
+};
+
+const walkThroughDesignTokenNodes = (
+  node: DesignTokenNode,
+  source: DesignTokenSpecification,
+  ancestors: ParentReference[],
+  mem: DesignTokenVariableSet,
+  nodeName?: string): DesignTokenVariableSet => {
+
+  if (isDesignTokenGroup(node)) {
+    Object.entries(node)
+      .filter(([tokenName, tokenNode]) => !tokenName.startsWith('$') && (isDesignToken(tokenNode) || isDesignTokenGroup(tokenNode)))
+      .forEach(([tokenName, tokenNode]) => walkThroughDesignTokenNodes(
+        tokenNode as DesignTokenGroup | DesignToken, source, nodeName ? [...ancestors, { name: nodeName, tokenNode: node }] : ancestors, mem, tokenName
+      ));
+  }
+
+  if (isDesignToken(node)) {
+    if (!nodeName) {
+      throw new Error('The first node of the Design Specification can not be a token');
```

Comment:
```suggestion
      throw new O3rCliError('The first node of the Design Specification can not be a token');
```
from `@o3r/schematics`

Reply:
Same reponse as for the CLI, I do not want to add a dep to `@o3r/schematics` in the `src` code